### ISSUE TITLE
Needless call of to_vec() on temporary Vec

### ISSUE THREAD

**Issue by Kixunil** [rust-lang/rust-clippy#7908]
### What it does

When `to_vec()` is called on temporary `Vec<_>` the lint should warn that `to_vec()` is not needed and causes extra allocation. (Suggesting removing it.)

### Categories

- Kind: performance

*What is the advantage of the recommended code over the original code*

- One less allocation
- More concise code (the returned value is already a `Vec<_>`)

### Drawbacks

None.

### Example

([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6db12f8e2e4f82c8bf1ce25a7eb90c8d))

```rust
fn create_vec() -> Vec<u8> {
    vec![1, 2, 42]
}

fn main() {
    println!("{:?}", create_vec().to_vec());
}
```

Could be written as:

```rust
fn create_vec() -> Vec<u8> {
    vec![1, 2, 42]
}

fn main() {
    println!("{:?}", create_vec());
}
```

**Comment by camsteffen** [comment]
This can be an enhancement of `redundant_clone`. And it should share logic with `implicit_clone`.

**Comment by tyilo** [comment]
> This can be an enhancement of `redundant_clone`. And it should share logic with `implicit_clone`.

As `redundant_clone` has been moved to the nursery, would it make sense to create a new lint for this?
I don't see how this could trigger false positives.
